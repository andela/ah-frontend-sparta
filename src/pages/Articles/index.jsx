import React from 'react';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import PropTypes from 'prop-types';
import countries from './countries';
import CreateArticles from '../../components/Articles/createArticles';
import { postArticle } from '../../actions/articlesActions';
import getArticleTags from '../../helpers/handleArticleTags';


const KeyCodes = {
  comma: 188,
  enter: 13,
};

const delimiters = [KeyCodes.comma, KeyCodes.enter];


export class Articles extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      tags: [
      ],
      title: '',
      description: '',
      body: '',
    };

    this.handleDelete = this.handleDelete.bind(this);
    this.handleAddition = this.handleAddition.bind(this);
    this.onSubmit = this.onSubmit.bind(this);
    this.handleChange = this.handleChange.bind(this);
  }

  onSubmit = (event) => {
    event.preventDefault();
    const { addArticle, history } = this.props;
    const articleTags = this.state.tags;
    const tags = getArticleTags(articleTags);
    const { title, description, body } = this.state;
    addArticle({
      title, description, body, tags,
    }, history);
  }


  handleChange = (event) => {
    event.preventDefault();
    this.setState({
      [event.target.name]: event.target.value,
    });
  }

  handleDelete(i) {
    const { tags } = this.state;
    this.setState({
      tags: tags.filter((tag, index) => index !== i),
    });
  }

  handleAddition(tag) {
    this.setState(state => ({ tags: [...state.tags, tag] }));
  }

  render() {
    const { title, description, body } = this.state;
    const { tags } = this.state;
    const suggestedContries = countries.map(country => ({
      id: country,
      text: country,
    }));
    return (
      <div>
        <CreateArticles
          tags={tags}
          suggestions={suggestedContries}
          delimiters={delimiters}
          handleDelete={this.handleDelete}
          handleAddition={this.handleAddition}
          onSubmit={this.onSubmit}
          handleChange={this.handleChange}
          title={title}
          description={description}
          body={body}
        />
      </div>
    );
  }
}
Articles.propTypes = {
  addArticle: PropTypes.func.isRequired,
  history: PropTypes.shape({
    push: PropTypes.func.isRequired,
  }).isRequired,
};
export default connect(null, { addArticle: postArticle })(withRouter(Articles));
