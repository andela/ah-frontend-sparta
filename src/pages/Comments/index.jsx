import React from 'react';
import Comment from '../../components/Comment';
import ReplyToArticle from '../../components/ReplyToArticle';

function structureComments(comments, showReplies, replyDisplayState, slug, postReplyToComment, deleteComment, updateComment, onChangeComment) {
  const jsxComments = [];
  for (let i = 0; i < comments.length; i += 1) {
    const comment = comments[i];
    jsxComments.push(<Comment key={comment.id} commentObject={comment} showReplies={showReplies} replyDisplayState={replyDisplayState} slug={slug} postReplyToComment={postReplyToComment} deleteComment={deleteComment} updateComment={updateComment} onChangeComment={onChangeComment} />);
  }
  return jsxComments;
}

const Comments = ({
  comments,
  showReplies,
  replyDisplayState,
  slug,
  postReplyToComment,
  onCommentChanged,
  postComment,
  deleteComment,
  updateComment,
  onChangeComment,
}) => (
  <div>
    <div style={{ marginTop: 100, marginBottom: 15 }}>
      <ReplyToArticle
        onCommentChanged={onCommentChanged}
        postComment={postComment}
      />
      <span style={{ fontSize: 28 }}>
        {comments.length}
        {' '}
        Comments
      </span>
    </div>
    {structureComments(comments, showReplies, replyDisplayState, slug, postReplyToComment, deleteComment, updateComment, onChangeComment)}
  </div>
);

export default Comments;
