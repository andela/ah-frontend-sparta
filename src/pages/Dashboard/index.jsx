import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import DashboardArticles from '../../components/Articles/DashboardArticles';
import { fetchArticles } from '../../actions/articlesActions';
import { getOriginal, getNext } from '../../actions/paginationActions';

import './Dashboard.scss';

const SearchAlert = ({ message, color }) => (
  <div className={`alert ${color} alert-dismissible fade show`} role="alert">
    {' '}
    {message}
    <button
      type="button"
      className="close"
      data-dismiss="alert"
      aria-label="Close"
    >
      <span aria-hidden="true">&times;</span>
    </button>
  </div>
);

export class Dashboard extends React.Component {
  componentDidMount() {
    const { getArticles, fetchOriginal } = this.props;
    getArticles();
    fetchOriginal();
  }

  fetchData = (event) => {
    event.preventDefault();
    const { fetchNext } = this.props;
    fetchNext(localStorage.next);
  };

  fetchPrevious = (event) => {
    event.preventDefault();
    const { fetchNext } = this.props;
    fetchNext(localStorage.previous);
  };

  render() {
    const {
      firstArticles, paginateArticles, pageState: { next, prev },
      articles,
      searchArticles,
    } = this.props;

    const searchResult = this.props.searchArticles.results;

    return (
      <div>
        {searchArticles && searchArticles.count === 0 ? (
          <SearchAlert
            message="OOPS!! The Article you are looking for could be not found. Try again!"
            color="alert-warning"
          />
        ) : (
          <SearchAlert
            message="Woow, your results were found"
            color="alert-success"
          />
        ) }
        {searchResult && searchResult.length !== 0 ? <DashboardArticles totalArticles={searchResult} /> : (paginateArticles.length === 0 ? <DashboardArticles firstArticles={firstArticles} totalArticles={articles} /> : <DashboardArticles paginateArticles={paginateArticles} firstArticles={firstArticles} totalArticles={articles} />)}
        <div className="container pagination-buttons article-btns">
          {prev ? (
            <button type="button" className="btn left pagButton" onClick={this.fetchPrevious}>
              Previous
            </button>
          ) : (
            <button type="button" className="btn left pagButton" disabled>
                Previous
            </button>
          )}
          { next ? (
            <button type="button" className="btn right pagButton" onClick={this.fetchData}>
              Next
            </button>
          ) : (
            <button type="button" className="btn right pagButton" disabled>
              Next
            </button>
          )}
        </div>
      </div>
    );
  }
}

Dashboard.defaultProps = {
  firstArticles: [],
  paginateArticles: [],
  searchArticles: '',
};
Dashboard.propTypes = {
  getArticles: PropTypes.func.isRequired,
  firstArticles: PropTypes.arrayOf(PropTypes.object),
  paginateArticles: PropTypes.arrayOf(PropTypes.object),
  fetchOriginal: PropTypes.func.isRequired,
  fetchNext: PropTypes.func.isRequired,
  searchArticles: PropTypes.arrayOf(PropTypes.object),
};

const mapStateToProps = state => ({
  articles: state.article.articles,
  paginateArticles: state.paginateArticles,
  firstArticles: state.firstArticles,
  pageState: state.pageState,
  searchArticles: state.search,
});
const mapDispatchToProps = dispatch => ({
  fetchOriginal: () => {
    dispatch(getOriginal());
  },
  fetchNext: (url) => {
    dispatch(getNext(url));
  },
  getArticles: () => {
    dispatch(fetchArticles());
  },
});

export default connect(mapStateToProps,
  mapDispatchToProps)(Dashboard);
