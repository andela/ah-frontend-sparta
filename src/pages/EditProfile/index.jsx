import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import firebase from 'firebase';
import EditProfileInput from '../../components/EditProfileInput';
import ProgressBar from '../../components/ProgressBar';
import {
  fetchProfileActions,
  updateProfileActions,
  saveImageActions,
} from '../../actions/profileActions';
import './EditProfile.scss';


export class EditProfile extends Component {
  state = {
    imageLoader: 0,
  };

  componentDidMount() {
    firebase.initializeApp({
      apiKey: process.env.apiKey,
      authDomain: process.env.authDomain,
      databaseURL: process.env.databaseURL,
      projectId: process.env.projectId,
      storageBucket: process.env.storageBucket,
      messagingSenderId: process.env.messagingSenderId,
      appId: process.env.appId,
    });

    const { fetchProfile } = this.props;
    const username = localStorage.getItem('username');
    this.setState({ username });
    fetchProfile(username);
  }

  onChange = (e) => {
    this.setState({
      [e.target.id]: e.target.value,
    });
  }

  handleSubmit = (e) => {
    const { updateProfile, history } = this.props;
    e.preventDefault();
    updateProfile(this.state, history);
  }

  uploadImage=(files) => {
    const { saveImage } = this.props;
    const { username } = this.state;
    const task = firebase
      .storage()
      .ref(`images/${files[0].name}`)
      .put(files[0]);
    task.on('state_changed', (snapshot) => {
      const progress = Math
        .round((snapshot.bytesTransferred / snapshot.totalBytes) * 100);
      this.setState({
        imageLoader: progress,
      });
    });
    task.then(() => {
      firebase.storage()
        .ref(`images/${files[0].name}`)
        .getDownloadURL().then((url) => {
          saveImage(url, username);
        });
    });
  }

  render() {
    const {
      profile: {
        username, firstname, lastname, bio, image,
      },
    } = this.props;
    const { imageLoader } = this.state;
    const email = localStorage.getItem('email');

    return (
      <div className="container">
        <div className="row">
          <div className="col" />
          <div className="title col">
            <h2>Edit Profile</h2>
          </div>
          <div className="col" />
        </div>
        <div className="row">
          <div className="col-3 cube">
            <div className="offset-5">
              {imageLoader !== 0
                ? <ProgressBar percentage={imageLoader} /> : <div />}
            </div>
            <div className="imageContainer offset-5">
              <img className="image" src={image} alt="profile pic" />
            </div>
            <div className="offset-6 changePic">
              <label htmlFor="file">
                  Change picture
                <input
                  className="inputFile"
                  onChange={e => this.uploadImage(e.target.files)}
                  type="file"
                  id="file"
                  accept="image/*"
                />
              </label>
            </div>
          </div>
          <form className="col form" onSubmit={this.handleSubmit}>
            <EditProfileInput
              label="Username"
              id="username"
              type="text"
              value={username}
              onChange={this.onChange}
            />
            <EditProfileInput
              label="Firstname"
              id="firstname"
              type="text"
              value={firstname}
              onChange={this.onChange}
            />
            <EditProfileInput
              label="Lastname"
              id="lastname"
              type="text"
              value={lastname}
              onChange={this.onChange}
            />
            <EditProfileInput
              label="email"
              id="email"
              type="email"
              value={email}
              disabled
            />
            <div className="col-6 bio">
              <label htmlFor="bio">
                Bio:
                <br />
                <textarea
                  className="EditProfileTextarea"
                  rows="7"
                  cols="50"
                  id="bio"
                  type="text"
                  defaultValue={bio}
                  onChange={this.onChange}
                />
              </label>
            </div>
            <button
              className="updateButton"
              type="submit"
            >
                UPDATE
            </button>
          </form>
        </div>
      </div>
    );
  }
}

export const mapStateToProps = state => (
  {
    profile: state.profile.user,
    imageLoading: state.profile.imageLoading,
  }
);

export default withRouter(
  connect(
    mapStateToProps,
    {
      updateProfile: updateProfileActions,
      fetchProfile: fetchProfileActions,
      saveImage: saveImageActions,
    },
  )(EditProfile),
);

EditProfile.propTypes = {
  profile: PropTypes.objectOf,
  history: PropTypes.objectOf,
  saveImage: PropTypes.func,
  fetchProfile: PropTypes.func,
  updateProfile: PropTypes.func,
};

EditProfile.defaultProps = {
  profile: {},
  history: {},
  saveImage: () => {},
  fetchProfile: () => {},
  updateProfile: () => {},
};
