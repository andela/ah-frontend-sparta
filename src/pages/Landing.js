import React from 'react';
import Navbar from '../components/Navbar';
import LandingPageWrapper from '../components/LandingPageWrapper';
import LandingImage from '../../images/landing_image.jpg';

import '../assets/scss/style.scss';


const Landing = () => (
    <div>
      <Navbar/>
      
      <div className="page-banner-image">
         <img className='LandingImage' src={LandingImage} alt="Landing image" />
      </div>
      <div className="small-screen">
        <div>
            <form>
              <input type="text" placeholder="SEARCH" />
            </form>
        </div>
        <div><a href="/login">Account</a></div>
      </div>
      <LandingPageWrapper  />
      <br></br>
      <br></br>
      <br></br>
    </div>
    
);

export default Landing;
