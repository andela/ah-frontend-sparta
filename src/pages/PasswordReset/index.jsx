import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import PasswordResetForm from '../../components/PasswordResetForm';
import { ChangePasswordActions } from
  '../../actions/resetActions/changePasswordActions';
import logo from '../../assets/images/logo.png';
import '../../assets/scss/Login.scss';

export class PasswordReset extends Component {
  constructor(props) {
    super(props);
    this.state = {
      newPassword: '',
      confirmPassword: '',
    };
  }

  onChange = (e) => {
    this.setState({ [e.target.name]: e.target.value });
  }

  handleSubmit = (event) => {
    event.preventDefault();
    const { newPassword, confirmPassword } = this.state;
    const { changePassword, match } = this.props;
    changePassword(
      newPassword,
      confirmPassword,
      match.params.token,
    );
  }

  render() {
    const { newPassword, confirmPassword } = this.state;

    return (
      <div className="outer">
        <div className="inner1">
          <img src={logo} alt="Logo" className="logo" />
          <div className="message">Welcome Back.</div>
        </div>
        <div className="inner2">
          <div>
            <div className="innerHalfContent">
              <PasswordResetForm
                onChange={this.onChange}
                handleSubmit={this.handleSubmit}
                confirmPassword={confirmPassword}
                newPassword={newPassword}
              />
            </div>
          </div>
        </div>
      </div>
    );
  }
}
PasswordReset.propTypes = {
  changePassword: PropTypes.func,
  match: PropTypes.shape(PropTypes.object.isRequired),
};

PasswordReset.defaultProps = {
  changePassword: () => { },
  match: () => { },
};
export default connect(null, { changePassword: ChangePasswordActions })(PasswordReset);
