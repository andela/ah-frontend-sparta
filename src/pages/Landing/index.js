import React from 'react';
import LandingPageWrapper from '../../components/LandingPageWrapper';
import LandingImage from '../../assets/images/landing_image.jpg';

import './Landing.scss';


const Landing = () => (
    <div>
      <div className="page-banner-image">
         <img className='LandingImage' src={LandingImage} alt="Landing image" />
      </div>
      <div className="small-screen">
        <div>
            <form>
              <input type="text" placeholder="SEARCH" />
            </form>
        </div>
        <div><a href="/login">Account</a></div>
      </div>
      <LandingPageWrapper  />
      <br></br>
      <br></br>
      <br></br>
    </div>
    
);

export default Landing;
