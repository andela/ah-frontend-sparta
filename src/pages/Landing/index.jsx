import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import ArticleContainer from '../../components/Articles/ArticleContainer';
import LandingImage from '../../assets/images/landing_image.jpg';
import { fetchArticles } from '../../actions/articlesActions';
import { getOriginal, getNext } from '../../actions/paginationActions';
import './Landing.scss';

export class Landing extends Component {
  componentDidMount() {
    const { getArticles, fetchOriginal } = this.props;
    getArticles();
    fetchOriginal();
  }

  fetchData = (event) => {
    event.preventDefault();
    const { fetchNext } = this.props;
    fetchNext(localStorage.next);
  };

  fetchPrevious = (event) => {
    event.preventDefault();
    const { fetchNext } = this.props;
    fetchNext(localStorage.previous);
  };

  render() {
    const { firstarticles, paginatearticles, pageState: { next, prev } } = this.props;
    return (
      <div>
        <div className="page-banner-image">
          <img className="LandingImage" src={LandingImage} alt="Landing" />
        </div>
        <div className="small-screen">
          <div>
            <form>
              <input type="text" placeholder="SEARCH" />
            </form>
          </div>
          <div>
            <a href="/login">Account</a>
          </div>
        </div>
        <div>
          <div>
            {paginatearticles.length === 0 ? <ArticleContainer articles={firstarticles} /> : <ArticleContainer articles={paginatearticles} />}
            <br />
            <br />
          </div>
        </div>
        <div className="container pagination-buttons">
          {prev ? (
            <button type="button" className="btn left pagButton" onClick={this.fetchPrevious}>
              Previous
            </button>
          ) : (
            <button type="button" className="btn left pagButton" disabled>
                Previous
            </button>
          )}
          {next ? (
            <button type="button" className="btn right pagButton" onClick={this.fetchData}>
              Next
            </button>
          ) : (
            <button type="button" className="btn right pagButton" disabled>
                Next
            </button>
          )}
        </div>
      </div>
    );
  }
}

Landing.defaultProps = {
  firstarticles: [],
  paginatearticles: [],
};
Landing.propTypes = {
  firstarticles: PropTypes.arrayOf(PropTypes.object),
  paginatearticles: PropTypes.arrayOf(PropTypes.object),
  getArticles: PropTypes.func.isRequired,
  fetchOriginal: PropTypes.func.isRequired,
  fetchNext: PropTypes.func.isRequired,
};

const mapStateToProps = state => ({
  paginatearticles: state.paginateArticles,
  firstarticles: state.firstArticles,
  pageState: state.pageState,
});
const mapDispatchToProps = dispatch => ({
  fetchOriginal: () => {
    dispatch(getOriginal());
  },
  fetchNext: (url) => {
    dispatch(getNext(url));
  },
  getArticles: () => {
    dispatch(fetchArticles());
  },
});
export default connect(mapStateToProps, mapDispatchToProps)(Landing);
