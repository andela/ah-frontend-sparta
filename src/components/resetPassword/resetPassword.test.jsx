import React from 'react';
import renderer from 'react-test-renderer';
import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { shallow } from 'enzyme';
import { ResetPasswordContainer, mapStateToProps } from
  '../../pages/PasswordReset/ResetPasswordContainer';
import { PasswordReset } from '../../pages/PasswordReset';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);
const initialState = {
  resetReducer: {
    message: '',
    error: '',
  },
};
const store = mockStore(initialState);
const props = {
  resetPassword: jest.fn(),
  changePassword: jest.fn(),
  match: jest.fn(),

};
const wrapper1 = renderer.create(
  <Provider store={store}>
    <ResetPasswordContainer {...props} />
  </Provider>,
);
const wrapper2 = renderer.create(
  <Provider store={store}>
    <PasswordReset {...props} />
    ,
  </Provider>,
);
it('should render without crashing', () => {
  expect(wrapper1).toMatchSnapshot();
});

it('should handle change', () => {
  const wrapper = shallow(
    <ResetPasswordContainer {...props} />,
  );
  const instance = wrapper.instance();
  instance.onChange({ target: { name: 'email', value: 'dorothy@gmail.com' } });
  expect(instance.state.email).toEqual('dorothy@gmail.com');
});

it('should map state to props', () => {
  const state = {
    email: '',
  };
  expect(mapStateToProps(state)).toEqual({
    email: '',
  });
});
it('should handle click', () => {
  const wrapper = shallow(
    <ResetPasswordContainer {...props} />,
  );
  const instance = wrapper.instance();
  instance.handleClick({ preventDefault: jest.fn() });
  expect(props.resetPassword).toHaveBeenCalled();
});

it('should have default props resetPassword', () => {
  expect(ResetPasswordContainer.defaultProps.resetPassword).toBeDefined();
});
it('should render without crashing ', () => {
  expect(wrapper2).toMatchSnapshot();
});
