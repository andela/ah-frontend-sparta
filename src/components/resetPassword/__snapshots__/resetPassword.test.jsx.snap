// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render without crashing  1`] = `
Array [
  <div
    className="outer"
  >
    <div
      className="inner1"
    >
      <img
        alt="Logo"
        className="logo"
        src={Object {}}
      />
      <div
        className="intro"
      >
        Welcome Back.
      </div>
    </div>
    <div
      className="inner2"
    >
      <div>
        <div
          className="innerHalfContent"
        >
          <div>
            <div
              className="signStr spacebottom"
            >
              <h4>
                Authors Haven Password Reset
              </h4>
            </div>
            <form
              onSubmit={[Function]}
            >
              <div
                className="inputContain"
              >
                <input
                  className="input"
                  name="newPassword"
                  onChange={[Function]}
                  placeholder="Enter New password"
                  type="password"
                  value=""
                />
              </div>
              <div
                className="inputContain"
              >
                <input
                  className="input"
                  name="confirmPassword"
                  onChange={[Function]}
                  placeholder="Confirm New password"
                  type="password"
                  value=""
                />
              </div>
              <button
                type="submit"
              >
                Reset
              </button>
            </form>
          </div>
        </div>
      </div>
    </div>
  </div>,
  ",",
]
`;

exports[`should render without crashing 1`] = `
<div>
  <div
    aria-hidden="true"
    aria-labelledby="exampleModalLabel"
    className="modal fade"
    id="forgotPassword"
    role="dialog"
    tabIndex="-1"
  >
    <div
      className="modal-dialog custom-dialog"
      role="document"
    >
      <div
        className="modal-content customise_model"
      >
        <div
          className="modal-header"
        >
          <h3
            className="modal-title"
            id="exampleModalLabel"
          >
            Authors Haven
          </h3>
          <button
            aria-label="Close"
            className="close"
            data-dismiss="modal"
            type="button"
          >
            <span
              aria-hidden="true"
            >
              Ã—
            </span>
          </button>
        </div>
        <div
          className="modal-body"
        >
          <div
            className="password_reset"
          >
            <div
              className="form-group"
            >
              <label
                className="col-form-label form_label"
                htmlFor="email"
                id="hfjfh"
              >
                <h5>
                  Forgotten Password?
                </h5>
                <h6
                  className="paragraph-infor"
                >
                  Please enter your Email address. A link with instructions will be sent to you to reset your Password
                </h6>
                <input
                  className="form-control"
                  id="recipient-email"
                  name="email"
                  onChange={[Function]}
                  placeholder="Enter your email"
                  required={true}
                  type="email"
                  value=""
                />
              </label>
            </div>
            <div
              className="modal-footer"
            >
              <button
                className="btn btn-primary send-button"
                data-dismiss="modal"
                onClick={[Function]}
                type="button"
              >
                Click to send link to email
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
